lib/hallux/internal/ordered_hash_bounds.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Hallux.Internal.OrderedHashBounds do
+  defstruct [:order, :min, :max]
+
+  alias Hallux.Internal.OrderedHashBounds.MinOrder
+  alias Hallux.Internal.OrderedHashBounds.Order
+
+  def hash(value) do
+    :crypto.hash(:sha, inspect(value))
```

Comment:
`inspect` depends on options of how to display a value. For example, with the default options, all lists of length 51 or greater would hash to the same value:

```elixir
iex(12)> str1 = inspect(Enum.to_list(1..51))                
"[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, ...]"
iex(13)> str2 = inspect(Enum.to_list(1..100))
"[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, ...]"
iex(14)> str1 == str2
true
```

It is safer to use `:erlang.term_to_binary`:

```elixir
iex(18)> hash1 = :erlang.term_to_binary(Enum.to_list(1..51))
<<131, 107, 0, 51, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17,
  18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36,
  37, 38, 39, 40, 41, 42, 43, 44, 45, 46, ...>>
iex(19)> hash2 = :erlang.term_to_binary(Enum.to_list(1..100))
<<131, 107, 0, 100, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17,
  18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36,
  37, 38, 39, 40, 41, 42, 43, 44, 45, 46, ...>>
iex(20)> hash1 == hash2                                      
false
```

Reply:
Thanks! I was looking for something like that and couldn't find it.

Reply:
I guess I still want to hash the binary first, otherwise comparisons will potentially be costly :thinking:

Reply:
Oh, I was suggesting only to replace `inspect` with `term_to_binary`, but not to use it as a hash (the variable name I used was misleading).

In essence: 
```suggestion
    :crypto.hash(:sha, :erlang.term_to_binary(value))
```